python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,258 @@
+            # e.g. this happens for SliceOutOfLowerBoundEndIndexModule
+            # i.e. when you slice down to a zero dim (by going out of bounds)
+            # NB this is actually extra bad because mlir::makeStridedLinearLayoutMap
+            # fails an assert
+            shape = map(lambda x: x or -1, arg.shape)
+            annotations.append(
+                # we need torch dtype here rather than np dtype in order to pass to ModuleBuilder
+                TorchTensorType(shape=tuple(shape), dtype=normalized_args[i].dtype)
+            )
+            tensor_args.append(arg)
+
+    # pull out tensor kwargs and put them in positional order
+    tensor_kwargs_flat = []
+    if unwrapped_normalized_kwargs:
+        tensor_kwargs = {}
+        arg_idxs = {
+            arg_name: i
+            for i, arg_name in enumerate(
+                [arg.name for arg in script_fun.schema.arguments]
+            )
+        }
+        for i, (kw, arg) in enumerate(unwrapped_normalized_kwargs.items()):
+            if isinstance(arg, np.ndarray):
+                # we need torch dtype here rather than np dtype in order to pass to ModuleBuilder
+                tensor_kwargs[arg_idxs[kw]] = (arg, normalized_kwargs[kw].dtype)
+
+        for i in range(len(tensor_kwargs)):
+            arg, arg_dtype = tensor_kwargs[i]
+            annotations.append(TorchTensorType(shape=tuple(arg.shape), dtype=arg_dtype))
+            tensor_kwargs_flat.append(arg)
+
+    return annotations, tensor_args, tensor_kwargs_flat
+
+
+def fix_inplace(schema, out, all_tensor_args):
+    qual, name = schema.name.split("::")
+    op = OP_REGISTRY.get_by_triple((qual, name, schema.overload_name))
+    if op.is_mutable:
+        if len(op.returns) > 1:
+            raise RuntimeError("TorchMLIR doesn't handle multiple aliased returns yet.")
+        else:
+            aliased_arg = next(
+                arg
+                for arg in op.arguments
+                if "alias_info" in arg and arg["alias_info"]["is_write"]
+            )
+            # TODO: over-engineered
+            assert (
+                "alias_info" in op.returns[0]
+                and op.returns[0]["alias_info"]["is_write"]
+                and len(op.returns[0]["alias_info"]["after"]) == 1
+                and op.returns[0]["alias_info"]["after"][0]
+            )
+            assert (
+                len(aliased_arg["alias_info"]["after"]) == 1
+                and aliased_arg["alias_info"]["after"][0]
+                == op.returns[0]["alias_info"]["after"][0]
+            )
+            np.copyto(all_tensor_args[0], out)
+
+    return out
+
+
+def try_torch_mlir_eager(op, args, kwargs, backend):
+    if hasattr(op, "op_name"):
+        op_name = op.op_name
+    elif hasattr(op, "__name__"):
+        # handle builtin_function_or_method
+        op_name = op.__name__
+    else:
+        raise RuntimeError(f"op {op} has no name")
+
+    if op_name == "detach":
+        raise RuntimeError("detaching")
```

Comment:
comment

Reply:
response below doesn't apply here; we don't handle `detach`ing because we don't handle anything related to wengert list (ie grad flow graph) construction.